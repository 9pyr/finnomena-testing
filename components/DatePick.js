import styles from 'styles/DatePicker.module.scss'

import React from 'react'
import { DatePicker } from 'antd'

const { RangePicker } = DatePicker
const DatePick = ({ value, onChange }) => {
  return (
    <div className={styles.date_picker}>
      <RangePicker value={value} onChange={onChange} />
    </div>
  )
}

export default DatePick
