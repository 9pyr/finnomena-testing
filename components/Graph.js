import styles from 'styles/Graph.module.scss'

import React from 'react'
import { Area } from '@ant-design/plots'

const Graph = ({ data = [] }) => {
  return (
    <div className={styles.graph_styled}>
      <Area
        renderer='svg'
        height={200}
        autoFit={false}
        data={data}
        xField={'thailand_fund_code'}
        yField={'nav_return'}
        xAxis={{ label: null, line: null }}
        yAxis={{ label: null, grid: null }}
        tooltip={{ customContent }}
        smooth={true}
        areaStyle={() => ({
          fill: 'l(270) 0:#ffffff 0.5:#7ec2f3 1:#1890ff',
        })}
      />
    </div>
  )
}

export default Graph

export const customContent = (title, data) => {
  const { value } = data?.[0] || {}
  return (
    <div className={styles.tooltip_custom}>
      <div>
        ชื่อกองทุน: <span className={styles.title}>{title}</span>
      </div>
      <div>
        ผลตอบแทน: <span className={styles.text_highlight}>{value}</span>
      </div>
    </div>
  )
}
