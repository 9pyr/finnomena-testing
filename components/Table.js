import React from 'react'
import dayjs from 'dayjs'
import { Table as AntdTable } from 'antd'

const Table = ({ dataSource, onChange }) => {
  return <AntdTable showSorterTooltip={false} columns={columns} dataSource={dataSource} onChange={onChange} rowKey={(record) => record.mstar_id} />
}

export default Table

const columns = [
  {
    title: 'ชื่อกองทุน',
    dataIndex: 'thailand_fund_code',
    key: 'thailand_fund_code',
    ellipsis: true,
    width: '30%',
  },
  {
    title: 'อันดับของกองทุน',
    dataIndex: 'nav_index',
    key: 'nav_index',
    sorter: (a, b) => a.nav_index - b.nav_index,
    ellipsis: true,
    width: '15%',
    align: 'center',
  },
  {
    title: 'เวลาที่ข้อมูลถูกอัพเดต',
    dataIndex: 'nav_date',
    key: 'nav_date',
    sorter: (a, b) => new Date(a.nav_date).getTime() - new Date(b.nav_date).getTime(),
    ellipsis: true,
    render: (text) => dayjs(text).format('DD MMMM YYYY'),
    width: '25%',
    align: 'center',
  },
  {
    title: 'ผลตอบแทน',
    dataIndex: 'nav_return',
    key: 'nav_return',
    sorter: (a, b) => a.nav_return - b.nav_return,
    ellipsis: true,
    width: '15%',
    align: 'right',
  },
  {
    title: 'ราคา',
    dataIndex: 'nav',
    key: 'nav',
    sorter: (a, b) => a.nav - b.nav,
    ellipsis: true,
    width: '15%',
    align: 'right',
  },
]
