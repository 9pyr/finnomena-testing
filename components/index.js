export { default as Table } from './Table'
export { default as Graph } from './Graph'
export { default as DatePick } from './DatePick'