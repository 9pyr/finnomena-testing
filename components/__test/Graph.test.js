import React from 'react'
import { Graph } from 'components'
import { render } from '@testing-library/react'
import { customContent } from 'components/Graph'

const defaultProps = {
  data: [
    {
      mstar_id: 'F00000ZMXD',
      thailand_fund_code: 'PRINCIPAL VNEQ-A',
      nav_return: 74.80117,
      nav: 14.154,
      nav_date: '2021-09-07T00:00:00.000Z',
      avg_return: 63.633823,
      nav_index: 1,
    },
    {
      mstar_id: 'F0000109C9',
      thailand_fund_code: 'PRINCIPAL VNEQ-I',
      nav_return: 74.79997,
      nav: 14.2656,
      nav_date: '2021-09-07T00:00:00.000Z',
      avg_return: 63.633823,
      nav_index: 2,
    },
    {
      mstar_id: 'F00000IT9T',
      thailand_fund_code: 'TMBOIL',
      nav_return: 71.99969,
      nav: 4.354,
      nav_date: '2021-09-06T00:00:00.000Z',
      avg_return: 65.394614,
      nav_index: 3,
    },
    {
      mstar_id: 'F000002SAM',
      thailand_fund_code: 'TISCOOIL',
      nav_return: 66.52248,
      nav: 4.3079,
      nav_date: '2021-09-06T00:00:00.000Z',
      avg_return: 65.394614,
      nav_index: 4,
    },
  ],
}
const setupValues = (props) => ({ ...defaultProps, ...props })

describe('Graph Component', () => {
  test('render has props', () => {
    const props = setupValues()
    const { container } = render(<Graph {...props} />)

    expect(container).toMatchSnapshot()
  })
  test('render is props undefined', () => {
    const { container } = render(<Graph />)

    expect(container).toMatchSnapshot()
  })

  test('render has props by customContent function', () => {
    const props = ['title_test', [{ value: 'value_test' }]]
    const { container } = render(customContent(...props))

    expect(container).toMatchSnapshot()
    expect(container.querySelector(`.title`).textContent).toMatch(RegExp(props[0]))
    expect(container.querySelector('.text_highlight').textContent).toMatch(RegExp(props[1].value))
  })
  test('render has props undefined by customContent function', () => {
    const props = []
    const { container } = render(customContent(...props))

    expect(container).toMatchSnapshot()
    expect(container.querySelector(`.title`).textContent).toBe('')
    expect(container.querySelector('.text_highlight').textContent).toBe('')
  })
})
