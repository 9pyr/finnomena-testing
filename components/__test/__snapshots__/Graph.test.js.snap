// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph Component render has props 1`] = `
<div>
  <div
    class="graph_styled"
  >
    <div
      data-chart-source-type="G2Plot"
      style="height: inherit;"
    >
      <div
        style="position:relative;"
      >
        <svg
          height="200"
          style="width: 400px; height: 200px; display: inline-block; vertical-align: middle;"
          width="400"
        >
          <defs
            id="defs_1"
          >
            <lineargradient
              id="gradient_1"
              x1="0"
              x2="-1.8369701987210294e-16"
              y1="1"
              y2="1.1102230246251565e-16"
            >
              <stop
                offset="0"
                stop-color="#ffffff"
              />
              <stop
                offset="0.5"
                stop-color="#7ec2f3"
              />
              <stop
                offset="1"
                stop-color="#1890ff"
              />
            </lineargradient>
            <clippath
              id="clip_1"
            >
              <path
                d="M -20,-20 l 0,0 l 0,240 l0 0 z"
                fill-opacity="1"
                opacity="1"
                stroke-opacity="1"
                stroke-width="1"
              />
            </clippath>
            <clippath
              id="clip_2"
            >
              <path
                d="M -20,-20 l 0,0 l 0,240 l0 0 z"
                fill-opacity="1"
                opacity="1"
                stroke-opacity="1"
                stroke-width="1"
              />
            </clippath>
          </defs>
          <g
            opacity="1"
          >
            <path
              d="M 0,0 l 400,0 l 0,200 l-400 0 z"
              fill="transparent"
              fill-opacity="1"
              opacity="1"
              stroke="none"
              stroke-opacity="1"
              stroke-width="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            >
              <g
                opacity="1"
              >
                <g
                  opacity="1"
                />
              </g>
              <g
                opacity="1"
              />
            </g>
            <g
              opacity="1"
            />
          </g>
          <g
            opacity="1"
          >
            <g
              clip-path="url(#clip_1)"
              opacity="1"
            >
              <path
                d="M 50 12.997074999999995C 50 12.997074999999995 110.04889006007913 12.997074999999995 150 13.00007500000001C 190.04889006007915 13.003082342483397 210.13687218137693 15.876191976407826 250 20.000775000000004C 290.13687218137693 24.153681976407825 350 33.69380000000001 350 33.69380000000001L 350 200C 350 200 290 200 250 200C 210 200 190 200 150 200C 110 200 50 200 50 200Z"
                fill="url(#gradient_1)"
                fill-opacity="0.25"
                opacity="1"
                stroke="none"
                stroke-opacity="1"
                stroke-width="1"
              />
            </g>
            <g
              clip-path="url(#clip_2)"
              opacity="1"
            >
              <path
                d="M 50 12.997074999999995C 50 12.997074999999995 110.04889006007913 12.997074999999995 150 13.00007500000001C 190.04889006007915 13.003082342483397 210.13687218137693 15.876191976407826 250 20.000775000000004C 290.13687218137693 24.153681976407825 350 33.69380000000001 350 33.69380000000001"
                fill="none"
                fill-opacity="1"
                opacity="1"
                stroke="#5B8FF9"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-opacity="1"
                stroke-width="2"
              />
            </g>
          </g>
          <g
            opacity="1"
          >
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Graph Component render has props by customContent function 1`] = `
<div>
  <div
    class="tooltip_custom"
  >
    <div>
      ชื่อกองทุน: 
      <span
        class="title"
      >
        title_test
      </span>
    </div>
    <div>
      ผลตอบแทน: 
      <span
        class="text_highlight"
      >
        value_test
      </span>
    </div>
  </div>
</div>
`;

exports[`Graph Component render has props undefined by customContent function 1`] = `
<div>
  <div
    class="tooltip_custom"
  >
    <div>
      ชื่อกองทุน: 
      <span
        class="title"
      />
    </div>
    <div>
      ผลตอบแทน: 
      <span
        class="text_highlight"
      />
    </div>
  </div>
</div>
`;

exports[`Graph Component render is props undefined 1`] = `
<div>
  <div
    class="graph_styled"
  >
    <div
      data-chart-source-type="G2Plot"
      style="height: inherit;"
    >
      <div
        style="position:relative;"
      >
        <svg
          height="200"
          style="width: 400px; height: 200px; display: inline-block; vertical-align: middle;"
          width="400"
        >
          <defs
            id="defs_2"
          >
            <clippath
              id="clip_3"
            >
              <path
                d="M -20,-20 l 0,0 l 0,240 l0 0 z"
                fill-opacity="1"
                opacity="1"
                stroke-opacity="1"
                stroke-width="1"
              />
            </clippath>
            <clippath
              id="clip_4"
            >
              <path
                d="M -20,-20 l 0,0 l 0,240 l0 0 z"
                fill-opacity="1"
                opacity="1"
                stroke-opacity="1"
                stroke-width="1"
              />
            </clippath>
          </defs>
          <g
            opacity="1"
          >
            <path
              d="M 0,0 l 400,0 l 0,200 l-400 0 z"
              fill="transparent"
              fill-opacity="1"
              opacity="1"
              stroke="none"
              stroke-opacity="1"
              stroke-width="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            >
              <g
                opacity="1"
              >
                <g
                  opacity="1"
                />
              </g>
              <g
                opacity="1"
              />
            </g>
            <g
              opacity="1"
            />
          </g>
          <g
            opacity="1"
          >
            <g
              clip-path="url(#clip_3)"
              opacity="1"
            />
            <g
              clip-path="url(#clip_4)"
              opacity="1"
            />
          </g>
          <g
            opacity="1"
          >
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
            <g
              opacity="1"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
